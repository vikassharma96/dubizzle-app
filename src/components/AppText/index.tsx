import React from 'react';
import {Text, TextProps} from 'react-native';
import colors from '../../styles/colors';
import defaultStyles from '../../styles/defaultStyles';

type AppTextProps = {
  color?: string;
} & TextProps;

const AppText = ({color, style, ...props}: AppTextProps) => {
  return (
    <Text
      style={[defaultStyles.text, {color: color ?? colors.text}, style]}
      {...props}
    />
  );
};

export default AppText;
