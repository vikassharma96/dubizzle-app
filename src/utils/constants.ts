// home screen constants
export const NO_DATA_AVAIL = 'No data available!';
export const DEFAULT_ITEM_COUNT = 10;
export const CREATED_AT = 'Created at:';
export const LAST_UPDATED = 'Last updated:';
export const FILE = 'File';
export const FILES = 'Files';
