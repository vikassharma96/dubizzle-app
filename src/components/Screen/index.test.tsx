import React from 'react';
import {Screen} from 'react-native-screens';
import renderer from 'react-test-renderer';
import {FILE} from '../../utils/constants';
import AppText from '../AppText';

describe('@components/Screen', () => {
  const component = (
    <Screen>
      <AppText>{FILE}</AppText>
    </Screen>
  );

  it('should match snapshots', () => {
    const componentTree = renderer.create(component).toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
