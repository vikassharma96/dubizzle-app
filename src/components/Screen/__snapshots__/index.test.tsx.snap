// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Screen should match snapshots 1`] = `
<RNSScreen
  collapsable={false}
  forwardedRef={[Function]}
  gestureResponseDistance={
    {
      "bottom": -1,
      "end": -1,
      "start": -1,
      "top": -1,
    }
  }
  style={{}}
>
  <Text
    style={
      [
        {
          "fontFamily": "Poppins-Medium",
          "fontSize": 12,
        },
        {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    File
  </Text>
</RNSScreen>
`;
